components/PriceRanges/PriceRanges.js

```diff
@@ -0,0 +1,98 @@
+var React = require('react');
+
+var Template = require('../Template');
+var PriceRangesForm = require('./PriceRangesForm');
+var cx = require('classnames');
+
+class PriceRanges extends React.Component {
+  getForm() {
+    return (
+      <PriceRangesForm
+        cssClasses={this.props.cssClasses}
+        labels={this.props.labels}
+        refine={this.refine.bind(this)}
+      />
+    );
+  }
+
+  getURLFromFacetValue(facetValue) {
+    if (!this.props.createURL) {
+      return '#';
+    }
+    return this.props.createURL(facetValue.from, facetValue.to, facetValue.isRefined);
+  }
+
+  getItemFromFacetValue(facetValue) {
+    let cssClassItem = cx(
+      this.props.cssClasses.item,
+      {[this.props.cssClasses.active]: facetValue.isRefined}
+    );
+    let url = this.getURLFromFacetValue(facetValue);
+    let key = facetValue.from + '_' + facetValue.to;
+    let handleClick = this.refine.bind(this, facetValue.from, facetValue.to);
+    return (
+      <div className={cssClassItem} key={key}>
+        <a
+          className={this.props.cssClasses.link}
+          href={url}
+          onClick={handleClick}
+          ref={key}
+        >
+          <Template data={facetValue} templateKey="item" {...this.props.templateProps} />
+        </a>
+      </div>
+    );
+  }
+
+  refine(from, to, event) {
+    event.preventDefault();
+    this.setState({
+      formFromValue: null,
+      formToValue: null
+    });
+    this.props.refine(from, to);
+  }
+
+  render() {
+    let form = this.getForm();
+    return (
+      <div>
+        <div className={this.props.cssClasses.list}>
+          {this.props.facetValues.map(facetValue => {
+            return this.getItemFromFacetValue(facetValue);
+          })}
+        </div>
+        {form}
```

Comment:
This is maybe due to the fact that most components we always only test the render() method even if that's some boilerplate to add. This way we can just refactor the whole component really easily.

In the end testing a React component output is just asserting some JSX on render(), the underlying implementation should not have to be tested when we can avoid it (sometimes for handlers we need to test it).